zeroize/src/lib.rs

```diff
@@ -795,6 +795,20 @@ unsafe fn volatile_set<T: Copy + Sized>(dst: *mut T, src: T, count: usize) {
     }
 }
 
+/// Zeroizes a flat type/struct. Only zeroizes the values that it owns, and it does not work on
+/// dynamically sized values or trait objects.
+#[inline(always)]
+pub fn zeroize_flat_type<T: Sized>(data: &mut T) {
```

Comment:
The function must be `unsafe`, not only zeroized value may be invalid for `T`, but you also do not have any "flateness" guarantees.

Reply:
It's also probably worth to make it accept `*mut T`.

Reply:
It has been adjusted